ownCloud File Provider/OCItem+FileProviderItem.m

```diff
@@ -55,16 +55,120 @@ - (NSString *)filename
 	return (self.name);
 }
 
++ (NSDictionary<NSString*, NSString*> *)overriddenUTIByMIMEType
+{
+	static dispatch_once_t onceToken;
+	static NSDictionary<NSString *, NSString *> *utiByMIMEType;
+
+	dispatch_once(&onceToken, ^{
+		utiByMIMEType = @{
+			@"application/vnd.oasis.opendocument.text" 			: @"org.oasis-open.opendocument.text",
+			@"application/vnd.oasis.opendocument.text-template" 		: @"org.oasis-open.opendocument.text-template",
+
+			@"application/vnd.oasis.opendocument.graphics" 			: @"org.oasis-open.opendocument.graphics",
+			@"application/vnd.oasis.opendocument.graphics-template" 	: @"org.oasis-open.opendocument.graphics-template",
+
+			@"application/vnd.oasis.opendocument.presentation" 		: @"org.oasis-open.opendocument.presentation",
+			@"application/vnd.oasis.opendocument.presentation-template" 	: @"org.oasis-open.opendocument.presentation-template",
+
+			@"application/vnd.oasis.opendocument.spreadsheet" 		: @"org.oasis-open.opendocument.spreadsheet",
+			@"application/vnd.oasis.opendocument.spreadsheet-template" 	: @"org.oasis-open.opendocument.spreadsheet-template",
+
+			@"application/vnd.oasis.opendocument.formula" 			: @"org.oasis-open.opendocument.formula",
+			@"application/vnd.oasis.opendocument.formula-template" 		: @"org.oasis-open.opendocument.formula-template"
+		};
+	});
+
+	return (utiByMIMEType);
+}
+
++ (NSDictionary<NSString*, NSString*> *)overriddenUTIBySuffix
+{
+	static dispatch_once_t onceToken;
+	static NSDictionary<NSString *, NSString *> *utiBySuffix;
+
+	dispatch_once(&onceToken, ^{
+		utiBySuffix = @{
+			// @"odt" : @"org.oasis-open.opendocument.text",
+			// @"ott" : @"org.oasis-open.opendocument.text-template",
+
+			// @"odg" : @"org.oasis-open.opendocument.graphics",
+			// @"otg" : @"org.oasis-open.opendocument.graphics-template",
+
+			// @"odp" : @"org.oasis-open.opendocument.presentation",
+			// @"otp" : @"org.oasis-open.opendocument.presentation-template",
+
+			// @"ods" : @"org.oasis-open.opendocument.spreadsheet",
+			// @"ots" : @"org.oasis-open.opendocument.spreadsheet-template",
```

Comment:
Why are some lines uncommented?

Reply:
As discussed in the meeting I added the comment explaining it.